### ISSUE TITLE
Create moderators group and permissions for publishing and approval

### ISSUE THREAD

**Issue by giohappy** [geosolutions-it/nexus-geonode#269]
The current a[dvanced workflow system](https://docs.geonode.org/en/master/admin/admin_panel/index.html#how-to-enable-the-advanced-workflow)  rely on admins and group managers to grant permissions for the approval and publishing of resources (`approve` and `publish` metadata flags).

We want to introduce the `moderators` group, which will be used as a "role" like we have done for [contributors](https://github.com/GeoNode/geonode/issues/7355).

Two new permissions approve_resourcebase and publish_resorcebase will be created and assigned to the `moderators` group.
Users assigned to this group will have the permissions to publish and/or approve any resource.

**Documentation for the advanced workflow must be updated accordingly**.

**Comment by marthamareal** [comment]
@giohappy Is this ready?

**Comment by giohappy** [comment]
@marthamareal I've set your PR as Draft, since it covers only a part of the task.

The idea is to adopt the new `approve_resources` and `publish_resources` (**notice the change in the name, since these are global permissions not resource specific**) inside the Advanced Workflow logic.
The current implemention, as descrived in the [dedicated docs](https://docs.geonode.org/en/master/admin/admin_panel/index.html#group-based-advanced-data-workflow).

In particular the new permissions come into play when [ADMIN_MODERATE_UPLOADS](https://github.com/GeoNode/geonode/blob/9a1784071ffc65a98ab19019b172564dff8a5b2e/geonode/settings.py?plain=1#L1838-L1842) is active. The comments above this settings explain the logic, which is based on the current owser being either the owner of the resource, a group manager of a user's group or an admin. This kind of conditional statements are spread all over the places where `ADMIN_MODERATE_UPLOADS` and `RESOURCE_PUBLISHING` are checked. 

The idea is to centralize the permission checks for the advanced workflow under the security module, and check wether a user has publishing / moderating permissions.

This method to check the moderator permissions will return true if the user:

 - is and admin
 - has metadata edit permissions on the specific resource (this [happens for admins and group managers](https://github.com/GeoNode/geonode/blob/9a1784071ffc65a98ab19019b172564dff8a5b2e/geonode/utils.py?plain=1#L1077-L1097))
 -  has the `approve_resources` permission

Something similar for the check of permissions to publish. User can publish if:

 - is an admin
 - has `publish_resources` permissions
 - he has the (already implemented) `publish_resourcebase` permission on the resource. This resource specific permission could probably be dropped since, AFAIK, only an admin can publish if `ADMIN_MODERATE_UPLOADS` is active.

Long story short, we want to continue the work of basing the security logic on groups and permissions, rather then specific users'attributes (group manager, memeber of a group, etc.).

**Comment by afabiani** [comment]
@giohappy nothing specific to add right now; just wanted to _highlight_ the critical points where the logic has been applied:

 - [geonode/utils](https://github.com/GeoNode/geonode/blob/9a1784071ffc65a98ab19019b172564dff8a5b2e/geonode/utils.py?plain=1#L1023)
 - [geonode/base/utils](https://github.com/GeoNode/geonode/blob/master/geonode/base/utils.py#L164)
 - [geonode/security/utils](https://github.com/GeoNode/geonode/blob/master/geonode/security/utils.py#L38)
 - [geonode/security/models](https://github.com/GeoNode/geonode/blob/9a1784071ffc65a98ab19019b172564dff8a5b2e/geonode/security/models.py#L142)

At client side there are few places where the _legacy_ `views` disable the form `checkboxes` accordingly, e.g.:

 - [geonode/layers/views](https://github.com/GeoNode/geonode/blob/9a1784071ffc65a98ab19019b172564dff8a5b2e/geonode/layers/views.py#L1116-L1131)

We will need to envisage how to manage such behavior with the new SPA based client.
Moreover we will have to implement again the approval workflow on client side, allowing it to _prevent_ resource editing whenever it has been `approved` or `published` from a _non-admin_ user; also in the case it is `apporved` but not yet `published` the _owner_ or _editor_ should be able to request the `admin` to revert the resource into `unpublished` state.

**Comment by giohappy** [comment]
>We will need to envisage how to manage such behavior with the new SPA based client.
Moreover we will have to implement again the approval workflow on client side, allowing it to prevent resource editing whenever it has been approved or published from a non-admin user; also in the case it is apporved but not yet published the owner or editor should be able to request the admin to revert the resource into unpublished state.

This is why I want to have everything based on permissions. The client shouldn't be aweare of the underlying workflows. 
The backend will send back the user permissions based on the underlying logic. So, for example, id the resource has been approved (and the advanced workflow is active) the returned permissions on the resource for any other user shouldn't include the edit permissions.
A new epic should be created to make the API aware of the advanced workflow scenario.
